### PR TITLE
Introduce MetallicRoughnessTechnique

### PR DESCRIPTION
**Pull request by mpersano**
Factor out some of the repeated code in MetallicRoughnessEffect.

### CODE REVIEW

File: src/core/metallicroughnesstechnique.cpp [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    , m_opaque(true)
+    , m_alphaCutoffEnabled(false)
+    , m_toneMappingAlgorithm(MetallicRoughnessEffect::Reinhard)
+    , m_invokeInitVertexShaderRequested(false)
+    , m_backFaceCulling(new QCullFace(this))
+    , m_metalRoughShaderBuilder(new QShaderProgramBuilder(this))
+    , m_metalRoughShader(new QShaderProgram(this))
+{
+    const auto enabledLayers = QStringList{ QStringLiteral("noBaseColorMap"),
+                                                QStringLiteral("noMetalRoughMap"),
```
Comment (lemirep) [inline src/core/metallicroughnesstechnique.cpp]: Is indent correct here ? usually clang-format doesn't behave like that

File: src/core/metallicroughnesstechnique_p.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
+    metallicroughnesstechnique_p.h
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2018-2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Paul Lemire <paul.lemire@kdab.com>
```
Comment (lemirep) [inline src/core/metallicroughnesstechnique_p.h]: thx :)

**Comment by jjcasmar** [comment]
I disagree in the way this patch refactor the technique. Before the patch, we only had one copy of each property, now we have 3 copies of each property, one for each technique. 
I also think its confusing and error prone to set the property for each technique, but only emit a signal from the GL3 technique. 
Isnt there a way to refactor the common bits of the technique, but leave properties on the effect?

**Comment by mpersano** [comment]
@jjcasmar Good points, updated the patch.

**Pull request merged**